posthog/temporal/tests/utils/models.py

```diff
@@ -0,0 +1,55 @@
+"""Test utilities to manipulate BatchExport* models."""
+import uuid
+
+import temporalio.client
+from asgiref.sync import sync_to_async
+
+from posthog.batch_exports.models import BatchExport, BatchExportBackfill, BatchExportDestination, BatchExportRun
+from posthog.batch_exports.service import sync_batch_export
+
+
+def create_batch_export(team_id: int, interval: str, name: str, destination_data: dict) -> BatchExport:
+    """Create a BatchExport and its underlying Schedule."""
+
+    destination = BatchExportDestination(**destination_data)
+    batch_export = BatchExport(team_id=team_id, destination=destination, interval=interval, name=name)
+
+    sync_batch_export(batch_export, created=True)
```

Comment:
I think we do have ids, just the transaction isn't committed so that it can be rolledback easily after the test. But yeah, this is a bit of django magic that I'm not super certain about neither.

Reply:
Ah, important to note that `id` is a UUID set by us. So django can know it even before `save`, as it doesn't need to check the database (like it would, for example, for a sequential id).

Reply:
Ahhh, I missed that it was a `UUIDModel` subclass. For normal autoincrement models I would assume they'd only get an `id` upon `save`. Thanks for explaining.